### PR TITLE
added: calculate grid parameters in ASMs3DLag without using spline object

### PR DESCRIPTION
**Pull request by akva2**

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/ASM/ASMs3DLag.C [hunk 1]
```diff
@@ -1054,3 +1054,39 @@ void ASMs3DLag::generateThreadGroups (char lIndex, bool, bool)
   threadGroupsFace[lIndex].calcGroups(d1,d2,1);
   threadGroupsFace[lIndex].applyMap(map);
 }
+
+
+bool ASMs3DLag::getGridParameters (RealArray& prm, int dir, int nSegPerSpan) const
+{
+  if (svol)
+     return this->ASMs3D::getGridParameters(prm, dir, nSegPerSpan);
```
Comment (kmokstad) [inline src/ASM/ASMs3DLag.C:1062]: Wrong indent (one space too much)

File: src/ASM/ASMs3DLag.C [hunk 2]
```diff
@@ -1054,3 +1054,39 @@ void ASMs3DLag::generateThreadGroups (char lIndex, bool, bool)
...
+
+  if (nSegPerSpan < 1)
+  {
+    std::cerr <<" *** ASMs3DLag::getGridParameters: Too few knot-span points "
+              << nSegPerSpan+1 <<" in direction "<< dir << std::endl;
+    return false;
+  }
+
+  int nel = dir == 0 ? (nx-1)/(p1-1) : (dir == 1 ? (ny-1)/(p2-1) : (nz-1)/(p3-1));
```
Comment (kmokstad) [inline src/ASM/ASMs3DLag.C:1071]: Prefer rename this variable to nel1 (number of elements in one direction). nel is confused as total number of elements in the patch (member in ASMbase).

File: src/ASM/ASMs3DLag.C (hunk 1 above)
Comment (akva2) [reply src/ASM/ASMs3DLag.C]: fixed.

File: src/ASM/ASMs3DLag.C (hunk 2 above)
Comment (akva2) [reply src/ASM/ASMs3DLag.C]: fixed.

**Pull request merged**